pkg/cosign/cosign.go

```diff
@@ -147,7 +148,7 @@ func buildCosignOptions(ctx context.Context, opts images.Options) (*cosign.Check
 			}
 
 			if opts.CertChain == "" {
-				cosignOpts.SigVerifier, err = signature.LoadVerifier(cert.PublicKey, crypto.SHA256)
+				cosignOpts.SigVerifier, err = signature.LoadVerifier(cert.PublicKey, signatureAlgorithm)
```

Comment:
It might be worth taking another look at this change.
From my understanding "signatureAlgorithm" is a property of [StaticKeyAttestor](https://htmlpreview.github.io/?https://github.com/kyverno/kyverno/blob/main/docs/user/crd/index.html#kyverno.io/v1.StaticKeyAttestor) alone and not part of [CertificateAttestor](https://htmlpreview.github.io/?https://github.com/kyverno/kyverno/blob/main/docs/user/crd/index.html#kyverno.io/v1.CertificateAttestor).
So passing a value set for "Key attestor" to "certificate attestor" doesn't seem proper to me. (and the value of "signatureAlgorithm" seems uninitialised for the certificate attestor code path).

Reply:
Oh I understand, you are right. It will be an empty in the case of a certificate attestor.

I think the right solution here would be to add signatureAlgorithm in certificateAttestor as well, to allow configuration of that value, since currently we are locking it to SHA256

Wdyt @johnthedev97

Reply:
agreed